test/lib/runtime/session_test.exs

```diff
@@ -426,13 +426,14 @@ defmodule Ask.SessionTest do
     assert_receive [:ask, ^test_channel, ^respondent_received, ^token, ReplyHelper.simple("Do you smoke?", "Do you smoke? Reply 1 for YES, 2 for NO")]
     assert 1 == respondent_received.stats |> Stats.attempts(:sms)
 
-    {:ok, session = %Session{token: token, respondent: respondent}, _, 3} = Session.timeout(session)
+    {:ok, session = %Session{token: token, respondent: respondent}, _, 3} = Session.timeout(session) # first retry
     refute_receive [:setup, _, _, _, _]
     assert_receive [:ask, ^test_channel, respondent_received, ^token, ReplyHelper.simple("Do you smoke?", "Do you smoke? Reply 1 for YES, 2 for NO")]
     assert respondent.id == respondent_received.id
     assert 2 == respondent_received.stats |> Stats.attempts(:sms)
 
-    result = Session.timeout(session)
+    {:ok, session, _, _} = Session.timeout(session) # second retry
```

Comment:
If I copy [this line](https://github.com/instedd/surveda/pull/1684/commits/7aa2680422ab1eafbcd834cd6cffb898407b991e#diff-9db734eb9cf2cf21192b04a2634fc7bdR431) immediately after the current line, when I run the test it fails with a `No message matching` error.

Do you know why it behaves differently in the 2nd retry than in the first one?

Reply:
Please discard the above question and consider the below suggestion.
It duplicates code, I know, but I think for common people like me it could help to make the test flow more understandable ¯\_(ツ)_/¯
And, yes, it could be much better to refactor it and make it much more understandable like we did with a lot of tests, but I didn't even try to get so far 😬